### PR TITLE
Fix comments being placed inside constructors

### PR DESCRIPTION
**Pull request by tadeuzagallo**
Fixes the issues described on #575

### CODE REVIEW

File: formatTest/typeCheckedTests/expected_output/reasonComments.re [hunk 1]
```diff
@@ -279,9 +279,8 @@ let result =
  */
 let res =
   switch (
-    X
-      /* Retain this */
-      (2, 3)
+    /* Retain this */
```
Comment (jordwalke) [inline formatTest/typeCheckedTests/expected_output/reasonComments.re]: Much better!

File: src/reason_pprint_ast.ml [hunk 2]
```diff
@@ -3829,7 +3829,7 @@ class printer  ()= object(self:'self)
                     (* TODO: Explicit arity *)
                     | `normal ->
                         let arityIsClear = isArityClear arityAttrs in
-                        self#constructor_expression ~arityIsClear stdAttrs (self#longident_loc li) eo
+                        self#constructor_expression ~arityIsClear stdAttrs (SourceMap (li.loc, self#longident_loc li)) eo
```
Comment (jordwalke) [inline src/reason_pprint_ast.ml]: Looks great.

**Comment by jordwalke** [comment]
Thank you, @tadeuzagallo.

**Comment by jordwalke** [comment]
Do the unit and idempotent tests still pass?

**Comment by tadeuzagallo** [comment]
Yes, `make test` was all green.

Also, I'd be happy to setup a CI for the repo if you want.

**Comment by jberdine** [comment]
nice!

File: src/reason_parser_message.ml [hunk 3]
```diff
@@ -10,29 +10,29 @@ let message =
         "<SYNTAX ERROR>\n"
```
Comment (yunxing) [inline src/reason_parser_message.ml]: Not related to this diff: 

The diff on this file is getting more annoying. I think we should just ignore this file and generate it on the fly.

**Comment by yunxing** [comment]
Looks great, thanks!

**Comment by tadeuzagallo** [comment]
Sorry, I'm new to the repo, should I just merge it? Or should I wait for a reviewer to press the button? 😄

**Comment by jordwalke** [comment]
Im merging!

**Comment by jordwalke** [comment]
Ah this has conflicts. Cannot merge!

**Comment by tadeuzagallo** [comment]
Oops, accidentally killed the PR by setting it to master 😞

**Comment by yunxing** [comment]
Thanks @tadeuzagallo. This should have already been fixed on master.e

**Pull request closed without merging**